jenkins-setup/setup.sh

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+echo "JENKINS_ROOT=$JENKINS_ROOT" >> "$GITHUB_ENV"
+if [[ "$JENKINS_PM_URL" =~ ^https://github.com/jenkinsci/plugin-installation-manager-tool.* ]]
```

Comment:
It should be clearly documented that `JENKINS_PM_VERSION`, `JENKINS_VERSION`, and `JFR_VERSION` are ignored if `JENKINS_PM_URL`, `JENKINS_CORE_URL`, or `JENKINS_JFR` are not set to default values, respectively.

Reply:
I agree with the need to add code comments / presentable documentation in such cases, otherwise it is easy to be gotcha's for the end user. These are really good examples of cases where a lack of documentation would result in countless hours spent on just figuring out why the code does not work as expected. Imagine how frustrating it would be.

Reply:
> It should be clearly documented that `JENKINS_PM_VERSION`, `JENKINS_VERSION`, and `JFR_VERSION` are ignored if `JENKINS_PM_URL`, `JENKINS_CORE_URL`, or `JENKINS_JFR` are not set to default values, respectively.

I added related comments here and also added in the inputs description tables.